easybuild/framework/easyconfig/style.py

```diff
@@ -0,0 +1,113 @@
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+Style tests for easyconfig files using pep8.
+
+@author: Ward Poelmans (Ghent University)
+"""
+
+import re
+from vsc.utils import fancylogger
+
+from easybuild.tools.utilities import only_if_module_is_available
+
+try:
+    import pep8
+except ImportError:
+    pass
+
+_log = fancylogger.getLogger('easyconfig.style', fname=False)
+
+
+# any function starting with _eb_check_ will be added to the tests
+# if the test number is added to the select list. The test number is
+# definied as AXXX where A is either E or W and XXX is a 3 digit number.
+# It should be mentioned in the docstring as a single word.
+# Read the pep8 docs to understand the arguments of these functions
+def _eb_check_trailing_whitespace(physical_line, lines, line_number, total_lines):
+    """
+    W299
+    Warn about trailing whitespace, expect for the description and comments.
+    This differs from the standard trailing whitespace check as that
+    will warn for any trailing whitespace.
+    """
+    comment_re = re.compile('^\s*#')
+    if comment_re.match(physical_line):
+        return None
+
+    result = pep8.trailing_whitespace(physical_line)
+    if result:
+        result = (result[0], result[1].replace("W291", "W299"))
+
+    # if the warning is about the multiline string of description
+    # we will not issue a warning
+    keys_re = re.compile("^(?P<key>[a-z_]+)\s*=\s*")
+
+    for line in reversed(lines[0:line_number]):
+        res = keys_re.match(line)
+        if res:
+            if res.group("key") == "description":
+                return None
+            else:
+                break
+
+    return result
+
+
+@only_if_module_is_available('pep8')
+def style_conformance(lst_easyconfigs, verbose=False):
+    """
+    Check the given list of easyconfigs for style
+    @param lst_easyconfigs list of file paths to easyconfigs
+    @param verbose print our statistics and be verbose about the errors and warning
+    @return the number of warnings and errors
+    """
+    # importing autopep8 changes some pep8 functions.
+    # We reload it to be sure to get the real pep8 functions.
+    reload(pep8)
+
+    # register the extra checks before using pep8:
+    # any function in this module starting with `_eb_check_` will be used.
+    cands = globals()
+    for check_function in sorted([cands[f] for f in cands if callable(cands[f]) and f.startswith('_eb_check_')]):
```

Comment:
sure, but `eb_check_` is what we call a 'magic string'

right now, it's sort of OK, since it's only used here (and in the function names, but there you can't use a constant anyway)

but we should definitely not start using it in other places, like log messages

in fact... I'd like to see a log message, or maybe even a print, for a list of checks functions being performed?